### ISSUE TITLE
[YSQL] We dont seem to be performing sequential reads on a table in parallel from 3 shards in case of hash partitioned table.

### ISSUE THREAD

**Issue by shantanugupta-yb** [yugabyte/yugabyte-db#16384]
Jira Link: [DB-5798](https://yugabyte.atlassian.net/browse/DB-5798)
### Description

Sequential scan to read 1M rows on both hash partitioned and range partitioned tables issue 977 storage read requests and takes around 1.2k ms.

Why can't we sequentially read table in parallel from 3 shards in case of hash partitioned table? 

Sequential read on range partitioned table with a primary key of range type is issuing 977 storage read requests and taking around ~1200 ms  and even with hash partitioned table with pkey of hash type we still issue 977 storage read request which is taking around same amount of time of ~1200ms.

When we pushdown a filter condition we read the output in parallel from all tablets so expectation is to read from all tablets in parallel for sequential scan as well.
```
yugabyte=# explain (analyse, verbose, dist) select Rangetbl_1.col_bigint_id_1 from Rangetbl_1;
                                                       QUERY PLAN                                                        
-------------------------------------------------------------------------------------------------------------------------
 Seq Scan on public.rangetbl_1  (cost=0.00..100.00 rows=1000 width=8) (actual time=1.651..1279.302 rows=1000000 loops=1)
   Output: col_bigint_id_1
   Storage Table Read Requests: 977
   Storage Table Execution Time: 1185.987 ms
 Planning Time: 0.036 ms
 Execution Time: 1330.509 ms
 Storage Read Requests: 977
 Storage Write Requests: 0
 Storage Execution Time: 1185.987 ms
 Peak Memory Usage: 0 kB
(10 rows)

Time: 1331.590 ms (00:01.332)
```

```
yugabyte=# explain (analyse, verbose, dist) select Hashtbl_1.col_bigint_id_1 from Hashtbl_1;
                                                          QUERY PLAN                                                          
------------------------------------------------------------------------------------------------------------------------------
 Seq Scan on public.hashtbl_1  (cost=0.00..100000.00 rows=1000000 width=8) (actual time=1.738..1377.510 rows=1000000 loops=1)
   Output: col_bigint_id_1
   Storage Table Read Requests: 978
   Storage Table Execution Time: 1271.986 ms
 Planning Time: 0.037 ms
 Execution Time: 1426.268 ms
 Storage Read Requests: 978
 Storage Write Requests: 0
 Storage Execution Time: 1271.986 ms
 Peak Memory Usage: 0 kB
(10 rows)
```

Time: 1427.084 ms (00:01.427)

### Warning: Please confirm that this issue does not contain any sensitive information

- [X] I confirm this issue does not contain any sensitive information.

[DB-5798]: https://yugabyte.atlassian.net/browse/DB-5798?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by sushantrmishra** [comment]
Closing this as duplicate of https://github.com/yugabyte/yugabyte-db/issues/16384, since the underlying issue is same -->  seq scan without filter  are not executed in parallel.
.
